### PR TITLE
[SQ bug] unifiy weight_amax for modules same input

### PR DESCRIPTION
**Pull request by xin3he**
## Type of Change

bug fix

## Description

unifiy weight_amax for modules same input, or qkv will get different scale and got wrong accuracy.

## Expected Behavior & Potential Risk

facebook/opt125m gets correct accuracy on ipex2.1, folding=False.

## How has this PR been tested?

local test

### CODE REVIEW

File: neural_compressor/adaptor/torch_utils/smooth_quant.py [hunk 1]
```diff
@@ -706,6 +709,16 @@ def transform(self, alpha=0.5, folding=False, percentile=99.999, op_types=['Line
             if need_calibration:  ##avoid multiple calibaration during tuning if the only difference is alpha
                 if self.insert_mul:
                     self.self_absorb_layers = self._get_all_layer_names()  # TODO: only support linear now.
+                    # fetch modules with the same input
+                    group_modules = self._trace(op_types, skip_unsupport=False)
+                    for k, v in group_modules.items():
+                        # use one input for qkv
+                        if len(v) > 1:
```
Comment (hshen14) [inline neural_compressor/adaptor/torch_utils/smooth_quant.py:716]: we don't need this checker, right? for loop can deal with it.

File: neural_compressor/adaptor/torch_utils/smooth_quant.py [hunk 2]
```diff
@@ -836,7 +849,7 @@ def _get_example_input(self):
 
         return self.example_inputs
 
-    def _trace(self, op_types):
+    def _trace(self, op_types, skip_unsupport=True):
```
Comment (hshen14) [inline neural_compressor/adaptor/torch_utils/smooth_quant.py:852]: skip_unsupported_modules?

**Comment by hshen14** [comment]
Please add UT and also fix the coverage gap.

File: neural_compressor/adaptor/torch_utils/smooth_quant.py (hunk 2 above)
Comment (xin3he) [reply neural_compressor/adaptor/torch_utils/smooth_quant.py:852]: okay

**Review by yintong-lu — APPROVED** [review]

File: neural_compressor/adaptor/torch_utils/smooth_quant.py [hunk 3]
```diff
@@ -706,6 +706,15 @@ def transform(self, alpha=0.5, folding=False, percentile=99.999, op_types=['Line
             if need_calibration:  ##avoid multiple calibaration during tuning if the only difference is alpha
                 if self.insert_mul:
                     self.self_absorb_layers = self._get_all_layer_names()  # TODO: only support linear now.
+                    # fetch modules with the same input
+                    group_modules = self._trace(op_types, skip_unsupported_layers=False)
```
Comment (wenhuach21) [inline neural_compressor/adaptor/torch_utils/smooth_quant.py:710]: Tracing is not so robust. if tracing has some issues, could we fallback to checking the calibration tensor though it's a little trick

**Review by wenhuach21 — APPROVED** [review]

**Pull request merged**